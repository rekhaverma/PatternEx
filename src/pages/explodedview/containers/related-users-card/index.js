export { default } from './related-users-card.container.jsx';
