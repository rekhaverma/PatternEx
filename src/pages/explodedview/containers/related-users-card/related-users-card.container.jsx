import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';

import NoData from 'components/no-data';

import { withSearchData } from '../../hoc';
import Section from '../../components/section';
import List, { RELATED_USERS } from '../../components/list';

export const RelatedUsers = (props) => {
  const { searchData, isSearchDataLoaded } = props;

  return (
    <Section size="small" title="Related Users" loaded={isSearchDataLoaded}>
      {Object.keys(searchData).includes('map_users') && searchData.map_users ? (
        <List
          type={RELATED_USERS}
          data={searchData.map_users}
          customClass={{
            base: 'details-card related-users',
            modifiers: ['+rows-highlight'],
          }}
        />
      ) : (
        <NoData
          intlId="global.nodata"
          intlDefault="There is no data to display"
          className="nodata"
          withIcon
        />
      )
      }

    </Section>
  );
};

RelatedUsers.displayName = 'DetailsCard';

RelatedUsers.propTypes = {
  'searchData': PropTypes.object,
  'isSearchDataLoaded': PropTypes.bool,
};

RelatedUsers.defaultProps = {
  'searchData': {},
  'isSearchDataLoaded': false,
};

const enhance = compose(withSearchData);

export default enhance(RelatedUsers);
