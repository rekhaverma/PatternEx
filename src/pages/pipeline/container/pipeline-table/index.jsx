export { default } from './pipeline-table.container';
