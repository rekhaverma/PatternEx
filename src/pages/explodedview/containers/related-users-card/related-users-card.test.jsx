import React from 'react';

import { RelatedUsers } from './related-users-card.container';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<RelatedUsers />', () => {
  describe('render: ', () => {
    it('should match with snapshot', () => {
      const props = {
        isSearchDataLoaded: true,
        searchData: {
          tor_distinct_sld_count: 0,
          src_ip_15_avg_packets: 2,
          src_ip_30_count_sent_bytes_e_packets_gt: 0,
          src_ip_54_count_low: 0,
          src_ip_25_count_repeat_bytes_sent: 2,
          tor_sld_length: 0,
          src_ip_34_avg_duration: 100.25,
          src_ip_29_count_received_bytes_e_packets_gt: 0,
          src_ip_06_avg_bytes: 268.75,
          src_ip_65_map_users: null,
          src_ip_33_std_interval: 19.70335560817553,
          src_ip_01_distinct_dest_locations: 1,
          src_ip_27_count_repeat_packets_sent: 4,
          src_ip_45_count_malware: 0,
          src_ip_46_count_web: 0,
          sld_digit_ratio: 0,
          src_ip_64_count_bytes_sent_1h_gt_100m: 1,
          user_tag: { 'label_id': null, 'system_tag': null, 'name': null, 'severity': 0, 'alert': null, 'type': null, 'id': null, 'description': null },
          src_ip_03_count_connections: 4,
          sessions_9000s_port: 0,
          src_ip_60_std_bytes_sent_1h: 214.8132869817155,
          src_ip_32_std_sessions_hour: 0.7993052538854536,
          fqdn_avg_num_special_chars: 0,
          predicted_tag_id: 'ebdad7ea-efa1-4c54-8c71-eff30b7aa4e1',
          src_ip_56_distinct_destinations: 0,
          srcip_hostname: null,
          nxdomain_digit_ratio: 0,
          predicted_prob_sum: 0,
          src_ip_12_std_bytes_sent: 2.8613807855648994,
          class_probabilities_map: { 'ebdad7ea-efa1-4c54-8c71-eff30b7aa4e1': 1, 'c56d395f-1369-40e3-b434-9e2d0bb9412f': 0 },
          src_ip_53_count_medium: 0,
          src_ip_31_counter_applications: 1,
          ad_tot_failed_logins: 0,
          ad_num_distinct_users_authenticated: 0,
          src_ip_09_std_bytes_received: 33.915335764223244,
          tor_sld_others_ratio: 0,
          srcip: '76.96.107.13',
          src_ip_21_std_packets_sent: 0,
          src_ip_35_count_destination_ips: 1,
          src_ip_18_std_packets_received: 0,
          ad_tot_logins: 0,
          nxdomain_sld_entropy: 0,
          src_ip_63_perc90_bytes_sent_1h: 0,
          src_ip_44_count_confirmed: 0,
          tor_char_continuity_rate: 0,
          max_dns_requests_hour: 0,
          num_nxdomain_queries: 0,
          src_ip_48_count_blocked_url: 0,
          src_ip_50_distinct_threat_ids: 0,
          ad_failed_login_ratio: null,
          src_ip_23_count_distinct_from_ports: 4,
          src_ip_58_max_bytes_sent_1h: 1075,
          src_ip_39_count_http_dataxfer: 0,
          src_ip_34_std_duration: 0.43301270189221863,
          tor_sessions_unusual_port: 0,
          src_ip_15_std_packets: 0,
          score: null,
          src_ip_62_perc50_bytes_sent_1h: 0,
          src_ip_06_std_bytes: 33.995404101142846,
          predicted_tag: {
            label_id: null,
            system_tag: true,
            name: 'Benign',
            severity: 0,
            alert: false,
            type: 'N',
            id: 'ebdad7ea-efa1-4c54-8c71-eff30b7aa4e1',
            description: 'System Benign tag',
          },
          num_dns_requests: 0,
          src_ip_33_avg_interval: -6.333333333333333,
          tor_ratio_candidate_ips: 0,
          tor_sld_vowel_ratio: 0,
          tor_sld_entropy: 0,
          src_ip_26_count_repeat_packets: 4,
          src_ip_51_count_critical: 0,
          ad_num_st: 0,
          src_ip_37_count_dns_tcp_gt_1mb: 0,
          src_ip_32_avg_sessions_hour: 0.16666666666666666,
          src_ip_61_perc10_bytes_sent_1h: 0,
          src_ip_52_count_high: 0,
          srcip_geo_location: { 'latitude': '37.751', 'city': null, 'state': null, 'longitude': '-97.822', 'country': 'United States' },
          src_ip_24_count_repeat_bytes: 0,
          src_ip_40_source_country: 'US',
          top_n_features: null,
          src_ip_28_count_unanswered_connections: 0,
          nxdomain_distinct_sld: 0,
          src_ip_43_tot_threat_sessions: 0,
          src_ip_12_avg_bytes_sent: 94.25,
          src_ip_09_avg_bytes_received: 174.5,
          src_ip_49_count_allowed: 0,
          src_ip_41_count_new_dips: 0,
          src_ip_21_avg_packets_sent: 1,
          src_ip_38_count_nonhttp_dataxfer: 0,
          predicted_prob: 1,
          fqdn_avg_length: 0,
          tor_max_sld_length: 0,
          tor_sld_digit_ratio: 0,
          src_ip_22_count_from_port_lt_1024: 0,
          src_ip_47_count_blocked: 0,
          src_ip_18_avg_packets_received: 1,
          src_ip_57_min_bytes_sent_1h: 0,
          tor_sessions_9000s_port: 0,
          min_nxdomain_sld_length: 0,
          src_ip_55_count_pcap_available: 0,
          src_ip_36_count_dns_udp_rule: 0,
          ts: 'Mon, 25 Jun 2018 00:00:00 -0000',
          src_ip_59_avg_bytes_sent_1h: 44.791666666666664,
          src_ip_42_count_new_ports: 0,
        },
      };
      const wrapper = shallow(<RelatedUsers {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
