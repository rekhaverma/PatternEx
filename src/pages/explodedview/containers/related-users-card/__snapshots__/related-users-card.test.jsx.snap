// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedUsers /> render:  should match with snapshot 1`] = `
<Section
  className=""
  dropDownOptions={Array []}
  dropDownState={false}
  dropdown={false}
  expandable={false}
  expandableView={null}
  filterFeatureList={[Function]}
  isDraggable={true}
  loaded={true}
  modalStyle={Object {}}
  onDropDownClick={[Function]}
  onDropDownSelection={[Function]}
  onModalOpen={[Function]}
  selectedDropDown={Array []}
  size="small"
  title="Related Users"
>
  <NoData
    className="nodata"
    intlDefault="There is no data to display"
    intlId="global.nodata"
    intlValues={Object {}}
    message=""
    withIcon={true}
  />
</Section>
`;
